artichoke-backend/src/value.rs

```diff
@@ -158,7 +158,8 @@ impl Value {
     ///
     /// This function can never fail.
     pub fn to_s_debug(&self) -> String {
-        format!("{}<{}>", self.ruby_type().class_name(), self.inspect())
+        let inspected_str = String::from_utf8_lossy(&self.inspect()).to_string();
+        format!("{}<{}>", self.ruby_type().class_name(), inspected_str)
```

Comment:
This code can be modified to avoid an allocation by assigning `self.inspect` to a temporary:

```rust
let inspect = self.inspect();
format!("{}<{}>", self.ruby_type().class_name(), String::from_utf8_lossy(inspect))
```